src/server/index.tsx

```diff
@@ -223,10 +223,17 @@ server.get("/*", async (req, res) => {
              <script defer src='/static/js/client.js'></script>
            </body>
          </html>
-`);
+  `);
   } catch (err) {
-    console.error(err);
-    return res.send(`404: ${removeAuthParam(err)}`);
+    var formatted: string;
+    if (err.status && err.message) {
+      res.status(err.status);
+      formatted = `${err.status}: ${err.message}`;
+    } else {
+      res.status(500);
+      formatted = err;
+    }
+    res.send(formatted);
```

Comment:
This could be improved further by displaying the error message as a full page with normal header and footer (if `site` was requested successfully). In case of error 401 it can also redirect to /login like some routes already do.

Reply:
The first part was mentioned in #982. If you don't do it as part of this PR, I could whip something up. I'm wondering: if the site returns a server error but the site is still fetched, is it a good idea to list the matrix user IDs of instance admins for users to reach out to for support? That information is already on the site response, so no further requests would be required.